server/game/cards/02_SHD/leaders/BoKatanKryzePrincessInExile.ts

```diff
@@ -0,0 +1,60 @@
+import AbilityHelper from '../../../AbilityHelper';
+import { LeaderUnitCard } from '../../../core/card/LeaderUnitCard';
+import { Trait, WildcardCardType } from '../../../core/Constants';
+import { StateWatcherRegistrar } from '../../../core/stateWatcher/StateWatcherRegistrar';
+import { AttacksThisPhaseWatcher } from '../../../stateWatchers/AttacksThisPhaseWatcher';
+
+export default class BoKatanKryzePrincessInExile extends LeaderUnitCard {
+    private attacksThisPhaseWatcher: AttacksThisPhaseWatcher;
+
+    protected override getImplementationId () {
+        return {
+            id: '7424360283',
+            internalName: 'bokatan-kryze#princess-in-exile',
+        };
+    }
+
+    protected override setupStateWatchers (registrar: StateWatcherRegistrar): void {
+        this.attacksThisPhaseWatcher = AbilityHelper.stateWatchers.attacksThisPhase(registrar, this);
+    }
+
+    protected override setupLeaderSideAbilities () {
+        this.addActionAbility({
+            title: 'If you attacked with a Mandalorian unit this phase, deal 1 damage to a unit',
+            cost: [AbilityHelper.costs.exhaustSelf()],
+            targetResolver: {
+                cardTypeFilter: WildcardCardType.Unit,
+                immediateEffect: AbilityHelper.immediateEffects.conditional({
+                    condition: (_) => this.attacksThisPhaseWatcher.getAttackers((attack) => attack.attacker.hasSomeTrait(Trait.Mandalorian)).length > 0,
+                    onTrue: AbilityHelper.immediateEffects.damage({ amount: 1 }),
+                    onFalse: AbilityHelper.immediateEffects.noAction(),
+                })
+            }
+        });
+    }
+
+    protected override setupLeaderUnitSideAbilities () {
+        this.addOnAttackAbility({
+            title: 'You may deal 1 damage to a unit. If you attacked with another Mandalorian unit this phase, you may deal 1 damage to a unit',
+            optional: true,
+            targetResolvers: {
+                firstDmg: {
+                    activePromptTitle: 'Deal 1 damage to a unit',
```

Comment:
The damage selector already has a built-in prompt so these are not necessary - it's not really important to tell the player that it is the second instance of damage, they will know that from context. We try to keep the targeting prompts as simple as possible

Reply:
Ahh, okay. That is in the test harness. Basically, every time `clickCard()` is called it checks to make sure that something about the prompt has changed to make sure that we did not click on a card that does nothing. In this case it is working right, it's just that the prompt for part 1 and part 2 is exactly the same so it gets confused.

You can use `clickCardNonChecking()` instead of `clickCard()` in cases where it is expected that the prompt will not change, please leave a comment explaining the reason